src/plugins/host-diagnostics/public/js/host-diagnostics.js

```diff
@@ -42,4 +49,33 @@
   HostDiagnostics.prototype.toggleDisplay = function toggleDisplay() {
   };
   window.Cockpit.plugins.push(HostDiagnostics);
+
+
+function takeOverConsole(){
+    var console = window.console
+    if (!console) return
+    function intercept(method){
+        var original = console[method]
+        console[method] = function(){
+            // do sneaky stuff
+            consoleRecorder.push({timestamp:Date.now(),msg:Array.prototype.slice.apply(arguments).join(' ')});
+            if (consoleRecorder.length>500){
+              consoleRecorder.shift();
+            }
+            if (original.apply){
+                // Do this for normal browsers
+                original.apply(console, arguments)
```

Comment:
Will double check, it should still pass through the original call, so that you see everything in console as well. Will try debugging with the intercept in place.